### PR TITLE
cut state into pieces

### PR DESCRIPTION
**Pull request by mkyc**
This PR is first in line of 2 or 3 to modify library to be more flexible in case of upgrades and communication between modules.

### CODE REVIEW

**Comment by seriva** [comment]
🎵Cut the state into piece... 🎵 This is our resort... 🎵

File: globals/interfaces.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package globals
+
+type Initializer interface {
+
+	// Init is responsible for building default version of structure to start from.
+	Init(moduleVersion string)
+}
+
+type Backuper interface { // TODO that name is not a real word
```
Comment (seriva) [inline globals/interfaces.go:9]: Backupper

**Review by seriva — COMMENTED** [review]
Like the new approach with the interfaces. Will be a lot cleaner once the other modules are converted as well.  

One more note outsite the scope of this review:

to.BooPtr -> to.Boo**L**Ptr

Also funny to see that Go is still as retarded as C/C++ when it comes to pointers to constants. Would think they would rethink some things with a new language.

File: globals/interfaces.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkyc) [reply globals/interfaces.go]: Done

**Review by erzetpe — APPROVED** [review]
Some small questions/remarks, but in general it is okay.

File: utils/load/load.go [hunk 2]
```diff
@@ -28,9 +27,6 @@ func State(path string) (*st.State, error) {
 		}
 
 		// TODO temporary code because of before mentioned issue
```
[the remark this answers is not in the document]
Comment (mkyc) [reply utils/load/load.go:29]: Good spot. Whole this package will be removed after all the modules are refactored. I'm not going to touch it now but in next PRs I will remove.

File: azbi/v0/config.go [hunk 3]
```diff
@@ -218,113 +110,139 @@ func (c *Config) isValid() error {
...
+	}
+	*c = *config
+	return nil
 }
 
-type OutputVmGroup struct {
-	Name *string    `json:"vm_group_name"`
-	Vms  []OutputVm `json:"vms"`
-}
+func (c *Config) UpgradeFunc(input map[string]interface{}) error {
```
[the remark this answers is not in the document]
Comment (mkyc) [reply azbi/v0/config.go:130]: That is convention that if something is providing function it is usually called `SomethingFunc`.

File: azbi/v0/config.go [hunk 4]
```diff
@@ -1,138 +1,31 @@
 package v0
 
 import (
-	"encoding/json"
 	"errors"
-	"fmt"
-
+	"github.com/epiphany-platform/e-structures/globals"
```
[the remark this answers is not in the document]
Comment (mkyc) [reply azbi/v0/config.go]: changed

File: azbi/v0/config.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (erzetpe) [reply azbi/v0/config.go:130]: Weird, but okay. :)

**Review by erzetpe — APPROVED** [review]

**Review by seriva — APPROVED** [review]

**Pull request merged**